### PR TITLE
feat: add sort to table grid

### PR DESCRIPTION
**Pull request by ashish-kale**
Add feature #2680

### CODE REVIEW

**Comment by adriannesoike** [comment]
@chasestarr are you troubleshooting the failed checks?

I'm also interested in this component, so let me know if you need help.

**Comment by chasestarr** [comment]
@ashish-kale you'll need to rebase and merge in https://github.com/ashish-kale/baseui/pull/1 for VRT tests

**Comment by ashish-kale** [comment]
@chasestarr merged snapshots as you requested. Also, added content to documentation

File: src/table-grid/__tests__/table-grid-sortable.scenario.js [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+import * as React from 'react';
+
+import {
+  StyledTable,
+  StyledBodyCell,
+  SortableHeadCell,
+  SORT_DIRECTION,
+} from '../index.js';
+
+export const name = 'table-grid-sortable';
```
Comment (chasestarr) [inline src/table-grid/__tests__/table-grid-sortable.scenario.js]: Ah sorry, I simplified scenario files last week. Please remove the name export

File: src/table-grid/__tests__/table-grid-sortable.scenario.js [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+              <StyledBodyCell key={cellIndex}>{cell}</StyledBodyCell>
+            ))}
+          </>
+        ))}
+      </StyledTable>
+    );
+  }
+}
+
+export const component = () => <SortableTable />;
```
Comment (chasestarr) [inline src/table-grid/__tests__/table-grid-sortable.scenario.js]: and make this the default export
```suggestion
export default SortableTable;
```

File: src/table-grid/__tests__/table-grid-sortable.scenario.js [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+import * as React from 'react';
+
+import {
+  StyledTable,
+  StyledBodyCell,
+  SortableHeadCell,
+  SORT_DIRECTION,
+} from '../index.js';
+
```
Comment (chasestarr) [inline src/table-grid/__tests__/table-grid-sortable.scenario.js]:
```suggestion
```

File: src/table-grid/__tests__/table-grid-sortable.scenario.js (hunk 2 above)
Comment (ashish-kale) [reply src/table-grid/__tests__/table-grid-sortable.scenario.js]: done

**Comment by chasestarr** [comment]
Thank You

**Pull request merged**